crates/wasmparser/src/resources.rs

```diff
@@ -213,10 +251,10 @@ pub trait WasmModuleResources {
     /// The global's value type must be canonicalized.
     fn global_at(&self, at: u32) -> Option<GlobalType>;
 
-    /// Returns the `FuncType` associated with the given type index.
+    /// Returns the `SubType` associated with the given type index.
     ///
-    /// The function type must be canonicalized.
-    fn func_type_at(&self, type_idx: u32) -> Option<&Self::FuncType>;
+    /// The sub type must be canonicalized.
+    fn sub_type_at(&self, type_index: u32) -> Option<&Self::SubType>;
```

Comment:
I feel like we might have talked about this in the past but I think it's probably time to remove these `*Resources` traits if we can since I don't think anything is actually using them. Especially as the type system gets more complicated with GC. Their intention was always to allow recording type information externally and then using that external type information to validate (which may not necessarily be wasmparser's type information), but I don't think there's any current users of that.

All-in-all I'd personally be more in favor of "just" returning `&wasmparser::SubType` here and nixing all the traits/impls.

Reply:
I thought maybe `wasmi` used it, but it seems not.

I'll do this in a follow up PR.